blog/2023-02-17-plugins.md

```diff
@@ -0,0 +1,127 @@
+  if (!process.env.PERMISSION_BOUNDARY_ARN) {throw new Error("env var PERMISSION_BOUNDARY_ARN not set")}
+  cdktf.Aspects.of(app).add(new PermissionBoundaryAspect(process.env.PERMISSION_BOUNDARY_ARN))
+}
+
+// Validate that all IAM roles have a permission boundary
+exports.validate = function (config) { }
+```
+
+So above we can see she created a new Aspect class that implements the `visit` method. Each node the aspect visits will be checked to determine if it is an IAM role and if so, set the permission boundary which was passed into the plugin through an environment variable `PERMISSION_BOUNDARY_ARN`
+
+Finally for her validate step, she will simply traverse the Terraform config for all IAM roles and check if the permission boundary is set. Even though her `preSynth` hook will have already done it job, she knows that `preSynth` is a mutable hook and that another plugin may have altered things after.
+
+```js
+// Validate that all IAM roles have a permission boundary
+exports.validate = function (config) {
+  for (const iamRole of Object.keys(config.resource.aws_iam_role)) {
+    const role = config.resource.aws_iam_role[iamRole];
+    if (!role.permission_boundary) {
+      throw new Error(`Role ${iamRole} does not have a permission boundary`);
+    }
+
+    if (role.permission_boundary !== process.env.PERMISSION_BOUNDARY_ARN) {
+      throw new Error(`Role ${iamRole} has incorrect permission boundary. Expected: ${process.env.PERMISSION_BOUNDARY_ARN} but got: ${role.permission_boundary}}`);
+    }
+  }
+}
+```
+
+Now Susan can use the plugin in her CD pipelines to ensure that all IAM roles have the correct permission boundary set, without imposing this non-functional requirement on the application developers. Susan will go on to write more plugins to help her organization meet their security and compliance requirements. She is no longer the underappreciated platform engineer we know from the beginning of our blog post, but rather a hero with her own corner office, private parking spot, and an on call pager that never goes off.
+
+:::note Susan Is A Fictional Character
+The out come of her success is purely speculative. Your company may not have corner offices so there is a chance you will have to just settle for the parking spot.
+:::
+
+## Ask Not What Your Plugin Can Do For You...
+
+This new plugin system is very exciting, and has a lot of possibilities. However, if it is to ever reach it full potential we need your help! If you have some ideas for useful plugins, or thoughts on additional hooks, or even just questions about how to make use of the plugin system, we want to hear from you! Open a pull request or an issue on our [GitHub](https://github.com/winglang/wing) also join our [community slack](https://t.winglang.io/slack) and let us know what you think.
+
+Want to read more about Wing plugins? Check out our [plugin documentation](https://docs.winglang.io/reference/compiler-plugins) for more information on the plugin system.
```

Comment:
Perhaps you can also link to the plugin examples and invite the reader to add more.